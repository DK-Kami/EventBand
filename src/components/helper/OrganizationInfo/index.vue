<template>
	<v-card class="custom-elevation">
		<v-layout column>
			<v-card-title>
				<v-layout 
					class="mx-2"
					wrap
				>
					<v-flex xs12 sm12 md2 lg2
						class="d-xs-flex d-md-none"
					>
						<v-layout :class="justify">
							<v-avatar :size="avatarSize">
								<v-img src="/Avatar.png"/>
							</v-avatar>
						</v-layout>
					</v-flex>
					<v-flex grow>
						<v-layout column>
							<span class="display-1 font-weight-medium"> {{organization.name}} </span>
							<span> Followers ( {{followersAmount}} ) </span>
							<span> Reputation ( {{organization.reputation}} ) </span>
							<v-rating
								:value="rating"
								:length="length"
								:size="ratingSize"
								background-color="accent lighten-3"
								color="accent lighten-1"
								half-increments
								readonly
								dense
							/>
						</v-layout>
					</v-flex>
					<v-flex xs12 sm12 md2 lg2
						class="d-none d-md-flex d-lg-flex"
					>
						<v-layout :class="justify">
							<v-avatar :size="avatarSize">
								<v-img :src="`https://event-band-api.ru:5000/static/organizations/${organization.logo}`"/>
							</v-avatar>
						</v-layout>
					</v-flex>
				</v-layout>
			</v-card-title>
			<v-divider/>
			<v-card-text>
				<v-layout class="title">
					{{organization.description}}
				</v-layout>
			</v-card-text>
		</v-layout>
	</v-card>
</template>

<script>
export default {
	name: 'OrganizationInfo',

	props: {
		organization: Object,
		followersAmount: Number,
		rating: Number,
	},

	data: () => ({
		length: 10,
	}),

	computed: {

		ratingSize() {
			switch (this.$vuetify.breakpoint.name) {
				case 'xs': return 16;
				case 'sm': return 32;
				case 'md': return 32;
				case 'lg': return 32;
				case 'xl': return 32;
			}
		},

		justify() {
			switch(this.$vuetify.breakpoint.name) {
				case 'xs': 
				case 'sm': return 'justify-center'
			}
		},

		avatarSize() {
			switch(this.$vuetify.breakpoint.name) {
				case 'xs': 
				case 'sm': return 256;
				case 'md':
				case 'lg': return 128;
				case 'xl': return 170;
				default: return 128;
			}
		}
	},
}
</script>