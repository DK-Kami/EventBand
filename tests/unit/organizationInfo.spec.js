import {
	mount,
	shallowMount
} from './index';

import OrganizationInfo from '@/components/helper/OrganizationInfo';

describe('Organization info - компонент для вывода информации об организации', () =>{
	let wrapper;

	beforeEach(() => {
		wrapper = mount(OrganizationInfo, {
			state: false,
			propsData: {
				"organization": {
					"id": 1,
					"uuid": "989713f3-2f0b-4a14-b863-d98b5c00f94e",
					"name": "Fregalo",
					"reputation": 9,
					"description": "\n          Normal fregal Sergeevich. Sounds honorable, looks stylish. Here, only fucked up in the head anime. Who are willing to sell their mother for ramen and a point for the opportunity to see waifa.\n          If you're not like that, just walk through and leave us alone. We don't touch you, don't touch us. Trust me, you'll be better off. We're fucked up tight, we can fuck you in all the holes if you just mention anime in our presence.\n        ",
					"logo": "fregalo.jpg",
					"createdAt": "2020-05-21T22:08:16.398Z",
					"updatedAt": "2020-05-21T22:08:16.398Z"
				},
				"followersAmount": 3,
				"rating": 8,
			},
		});
	});

	it('Делаем snapshot', () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('Является ли компонентом Vue', () => {
		expect(wrapper.isVueInstance()).toBeTruthy();
	});

	it('Card имеет заголовок', () => {
		expect(wrapper.find('.v-card__title').exists()).toBeTruthy();
	});

	it('Card имеет text', () => {
		expect(wrapper.find('.v-card__text').exists()).toBeTruthy();
	});

	it('Имеет аватар ', () => {
		expect(wrapper.find('.v-avatar').exists()).toBeTruthy();
	});
	
	
});