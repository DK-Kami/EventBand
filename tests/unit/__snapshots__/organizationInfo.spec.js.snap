// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organization info - компонент для вывода информации об организации Делаем snapshot 1`] = `
<div class="custom-elevation v-card v-sheet theme--light">
  <div class="layout column">
    <div class="v-card__title">
      <div class="layout mx-2 wrap">
        <div class="flex d-xs-flex d-md-none xs12 sm12 md2 lg2">
          <div class="layout">
            <div class="v-avatar" style="height: 128px; min-width: 128px; width: 128px;">
              <div class="v-responsive v-image">
                <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
                <div class="v-responsive__content"></div>
              </div>
            </div>
          </div>
        </div>
        <div class="flex grow">
          <div class="layout column"><span class="display-1 font-weight-medium"> Fregalo </span> <span> Followers ( 3 ) </span> <span> Reputation ( 9 ) </span>
            <div class="v-rating v-rating--readonly v-rating--dense"><button type="button" tabindex="-1" class="v-icon notranslate v-icon--link mdi mdi-star theme--light accent--text text--lighten-1" style="font-size: 32px;"></button><button type="button" tabindex="-1" class="v-icon notranslate v-icon--link mdi mdi-star theme--light accent--text text--lighten-1" style="font-size: 32px;"></button><button type="button" tabindex="-1" class="v-icon notranslate v-icon--link mdi mdi-star theme--light accent--text text--lighten-1" style="font-size: 32px;"></button><button type="button" tabindex="-1" class="v-icon notranslate v-icon--link mdi mdi-star theme--light accent--text text--lighten-1" style="font-size: 32px;"></button><button type="button" tabindex="-1" class="v-icon notranslate v-icon--link mdi mdi-star theme--light accent--text text--lighten-1" style="font-size: 32px;"></button><button type="button" tabindex="-1" class="v-icon notranslate v-icon--link mdi mdi-star theme--light accent--text text--lighten-1" style="font-size: 32px;"></button><button type="button" tabindex="-1" class="v-icon notranslate v-icon--link mdi mdi-star theme--light accent--text text--lighten-1" style="font-size: 32px;"></button><button type="button" tabindex="-1" class="v-icon notranslate v-icon--link mdi mdi-star theme--light accent--text text--lighten-1" style="font-size: 32px;"></button><button type="button" tabindex="-1" class="v-icon notranslate v-icon--link mdi mdi-star-outline theme--light accent--text text--lighten-3" style="font-size: 32px;"></button><button type="button" tabindex="-1" class="v-icon notranslate v-icon--link mdi mdi-star-outline theme--light accent--text text--lighten-3" style="font-size: 32px;"></button></div>
          </div>
        </div>
        <div class="flex d-none d-md-flex d-lg-flex xs12 sm12 md2 lg2">
          <div class="layout">
            <div class="v-avatar" style="height: 128px; min-width: 128px; width: 128px;">
              <div class="v-responsive v-image">
                <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
                <div class="v-responsive__content"></div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    <div class="v-card__text">
      <div class="layout title">

        Normal fregal Sergeevich. Sounds honorable, looks stylish. Here, only fucked up in the head anime. Who are willing to sell their mother for ramen and a point for the opportunity to see waifa.
        If you're not like that, just walk through and leave us alone. We don't touch you, don't touch us. Trust me, you'll be better off. We're fucked up tight, we can fuck you in all the holes if you just mention anime in our presence.

      </div>
    </div>
  </div>
</div>
`;
